pkgs/development/libraries/science/math/or-tools/default.nix

```diff
@@ -1,109 +1,95 @@
-  # the available solvers: https://github.com/google/or-tools/blob/b77bd3ac69b7f3bb02f55b7bab6cbb4bab3917f2/ortools/linear_solver/linear_solver.cc#L427
-  # We don't compile with SCIP because it does not have an open source license.
-  # See https://github.com/google/or-tools/issues/2395
-  preBuild = ''
-    for file in ortools/linear_solver/samples/*.cc; do
-      if grep -q SCIP_MIXED_INTEGER_PROGRAMMING $file; then
-        substituteInPlace $file --replace SCIP_MIXED_INTEGER_PROGRAMMING CBC_MIXED_INTEGER_PROGRAMMING
-      fi;
-    done
-
-    substituteInPlace ortools/linear_solver/samples/simple_mip_program.cc \
-      --replace 'SCIP' 'CBC'
-  '';
-  makeFlags = [
-    "prefix=${placeholder "out"}"
-    "PROTOBUF_PYTHON_DESC=${python.pkgs.protobuf}/${python.sitePackages}/google/protobuf/descriptor_pb2.py"
+  patches = [
+    # Disable test that requires external input: https://github.com/google/or-tools/issues/3429
+    (fetchpatch {
+      url = "https://github.com/google/or-tools/commit/7072ae92ec204afcbfce17d5360a5884c136ce90.patch";
+      hash = "sha256-iWE+atp308q7pC1L1FD6sK8LvWchZ3ofxvXssguozbM=";
+    })
+    # Fix test that broke in parallel builds: https://github.com/google/or-tools/issues/3461
+    (fetchpatch {
+      url = "https://github.com/google/or-tools/commit/a26602f24781e7bfcc39612568aa9f4010bb9736.patch";
+      hash = "sha256-gM0rW0xRXMYaCwltPK0ih5mdo3HtX6mKltJDHe4gbLc=";
+    })
   ];
-  buildFlags = [ "cc" "pypi_archive" ];
-
-  doCheck = true;
-  checkTarget = "test_cc";
-
-  installTargets = [ "install_cc" ];
-  # The upstream install_python target installs to $HOME.
-  postInstall = ''
-    mkdir -p "$python/${python.sitePackages}"
-    (cd temp_python/ortools; PYTHONPATH="$python/${python.sitePackages}:$PYTHONPATH" python setup.py install '--prefix=$python')
-  '';
-
-  # protobuf generation is not thread safe
-  enableParallelBuilding = false;
 
+  cmakeFlags = [
+    "-DBUILD_DEPS=OFF"
+    "-DBUILD_PYTHON=ON"
+    "-DBUILD_pybind11=OFF"
+    "-DFETCH_PYTHON_DEPS=OFF"
+    "-DUSE_GLPK=ON"
+    "-DUSE_SCIP=OFF"
+  ];
   nativeBuildInputs = [
     cmake
-    lsb-release
-    swig4
-    which
     ensureNewerSourcesForZipFilesHook
-    python.pkgs.setuptools
-    python.pkgs.wheel
+    pkg-config
+    swig4
+    unzip
   ];
   buildInputs = [
-    zlib
     bzip2
+    cbc
+    eigen
+    glpk
     python
+    python.pkgs.absl-py
+    python.pkgs.matplotlib
+    python.pkgs.mypy-protobuf
+    python.pkgs.pandas
+    python.pkgs.pip
+    python.pkgs.pybind11
+    python.pkgs.setuptools
+    python.pkgs.virtualenv
+    python.pkgs.wheel
+    re2
+    zlib
   ];
   propagatedBuildInputs = [
     abseil-cpp
     protobuf
-
     python.pkgs.protobuf
-    python.pkgs.six
-    python.pkgs.absl-py
-    python.pkgs.mypy-protobuf
+    python.pkgs.numpy
   ];
 
+  doCheck = true;
+
+  installPhase = ''
+    cmake . -DBUILD_EXAMPLES=OFF -DBUILD_PYTHON=OFF -DBUILD_SAMPLES=OFF
+    cmake --install .
```

Comment:
Do we compile here? That should happen in buildPhase.

Reply:
No, this is a configure step and an install step; no compiling happens here.

The reason for this extra configure step is to (a) prevent the installer from littering `$out/bin` with sample programs that only really function as tests, and (b) to disable the upstream installation of a zipped Python egg that can’t be imported with our Python setup.

Reply:
I think adding a short comment about that would help if that question comes up again in the future.